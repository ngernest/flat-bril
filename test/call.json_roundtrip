{
  "functions": [
    {
      "args": [],
      "instrs": [
        {
          "dest": "v",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "args": [],
          "funcs": [
            "print4"
          ],
          "labels": [],
          "op": "call"
        },
        {
          "args": [
            "v"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        }
      ],
      "name": "main"
    },
    {
      "args": [],
      "instrs": [
        {
          "dest": "v",
          "op": "const",
          "type": "int",
          "value": 4
        },
        {
          "args": [
            "v"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        },
        {
          "args": [],
          "funcs": [],
          "labels": [],
          "op": "ret"
        }
      ],
      "name": "print4"
    }
  ]
}
