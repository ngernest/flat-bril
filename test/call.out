main
op: const	dest: v 	type: int	value: 2	
op: call 	funcs: ["print4"]
op: print	args: ["v"]	
print4
op: const	dest: v 	type: int	value: 4	
op: print	args: ["v"]	
op: ret  	
